src/read/pipelining.rs

```diff
@@ -0,0 +1,1209 @@
+//! Pipelined extraction into a filesystem directory.
+
+pub mod path_splitting {
+    use displaydoc::Display;
+    use thiserror::Error;
+
+    use std::collections::BTreeMap;
+
+    use crate::spec::is_dir;
+
+    /// Errors encountered during path splitting.
+    #[derive(Debug, Display, Error)]
+    pub enum PathSplitError {
+        /// entry path format error: {0:?}
+        PathFormat(String),
+        /// file and directory paths overlapped: {0:?}
+        FileDirOverlap(String),
+    }
+
+    fn split_by_separator(entry_path: &str) -> Result<impl Iterator<Item = &str>, PathSplitError> {
```

Comment:
The ZIP spec states (APPNOTE 4.4.17) that file paths are in Unix format, and Unix filesystems treat a backslash as a normal character. Thus they should be allowed on Unix and replaced with `\u{fffd}` on Windows.

Reply:
Thanks so much for that reference!! Will fix this to conform to that.